bertopic/_bertopic.py

```diff
@@ -3980,18 +3980,29 @@ def _extract_topics(
             embeddings: The document embeddings
             mappings: The mappings from topic to word
             verbose: Whether to log the process of extracting topics
+            calculate_representation: Whether to extract the topic representations
 
         Returns:
             c_tf_idf: The resulting matrix giving a value (importance score) for each word per topic
         """
         if verbose:
-            logger.info("Representation - Extracting topics from clusters using representation models.")
+            action = "Representation" if calculate_representation else "Topics"
+            logger.info(
+                f"{action} - Extracting topics from clusters{' using representation models' if calculate_representation else ''}."
+            )
+
```

Comment:
Just to be sure I understood this correctly, are the two options as follows:

> "Representation - Extracting topics from clusters using representation models"
 
Does this refer to the case where **all** representations are calculated (and not just c-TF-IDF)?

> "Topics - Extracting topics from clusters"

Does this then refer to the case where **only** c-TF-IDF is calculate?

Also, perhaps it would be nice to create a variable for the second if/else statement the same way you did for the first. 

```python 
method = ' using representation models' if calculate_representation else ''
logger.info(f"{action} - Extracting topics from clusters{method}."
```

Reply:
Indeed, the `"Topics - Extracting topics from clusters"` refers to the case where only the c-TF-IDF is calculated.
And I would suggest changing it to `"Extracting base representation from bag of words"`

I can make the change suggested, but please check my comment [below](https://github.com/MaartenGr/BERTopic/pull/2180#issuecomment-2425958936).